src/js/helpers/nodesHelper.js

```diff
@@ -14,65 +14,79 @@
  * limitations under the License.
  */
 
-export const knotxNodes = () => {
-  const rootElem = document.body;
-  const COMMENTS = [];
-  const COMMENT_NODE_CODE = 8;
-  const HTML_NODE_CODE = 1;
-  // Fourth argument, which is actually obsolete according to the DOM4 standard, is required in IE 11
-  const iterator = document.createNodeIterator(
-    rootElem,
-    NodeFilter.SHOW_ALL,
-  );
-
-  let curNode;
-  let isBetweenComments = false;
-  let curComment = '';
-  // eslint-disable-next-line no-cond-assign
-  while ((curNode = iterator.nextNode())) {
-    if (
-      curNode.nodeType === COMMENT_NODE_CODE
-          && curNode.data.indexOf('data-knotx-id') !== -1
-    ) {
-      isBetweenComments = !isBetweenComments;
-      curComment = isBetweenComments ? curNode.data.match(/"([^']+)"/)[1] : '';
-    }
+export const isFragmentBoundary = (node) => node.nodeType === Node.COMMENT_NODE
+  && /data-knotx-id="(.*)"/.test(node.data);
 
-    if (
-      isBetweenComments
-          && curNode.nodeType !== COMMENT_NODE_CODE
-          && curNode.nodeType === HTML_NODE_CODE
-    ) {
-      if (!COMMENTS.length) {
-        curNode.dataset.knotxId = curComment;
-        COMMENTS.push(curNode);
-      } else if (!COMMENTS[COMMENTS.length - 1].contains(curNode)) {
-        curNode.dataset.knotxId = curComment;
-        COMMENTS.push(curNode);
-      }
+export const isCorrectFragmentNode = (node) => node.nodeType === Node.ELEMENT_NODE
```

Comment:
Can we have incorrect fragment nodes?
```suggestion
export const isFragmentNode = (node) => node.nodeType === Node.ELEMENT_NODE
```